arroyo-datastream/src/lib.rs

```diff
@@ -1622,9 +1636,75 @@ impl Program {
                             flat_map: Box::new(#func),
                         })
                     }
-                },
+                }
+                Operator::AsyncMapOperator {
+                    name,
+                    ordered,
+                    fn_name,
+                    defs, args,
+                    null_output_assignments,
+                    output_assignments,
+                    null_handlers,
+                    return_nullable,
+                    timeout_seconds,
+                    max_concurrency
+                } => {
+                    let in_k = parse_type(&input.unwrap().weight().key);
+                    let in_t = parse_type(&input.unwrap().weight().value);
+                    let out_t = parse_type(&output.unwrap().weight().value);
+
+                    let defs: TokenStream = parse_str(defs).expect("defs");
+                    let args: TokenStream = parse_str(args).expect("args");
+                    let fn_name: Ident = parse_str(fn_name).expect("fn_name");
+                    let null_output_assignments: TokenStream = parse_str(null_output_assignments).expect("null_output_assignments");
+                    let output_assignments: TokenStream = parse_str(output_assignments).expect("output_assignments");
+                    let null_handlers: TokenStream = parse_str(null_handlers).expect("null_handlers");
+
+                    let null_output = if *return_nullable {
+                        quote! {
+                            let null_output = (
+                                index.clone(),
+                                Ok(#out_t {
+                                    #null_output_assignments
+                                })
+                            );
+                        }
+                    } else {
+                        quote! ()
+                    };
+
+                    let udf_wrapper = quote!({
```

Comment:
I think there's too much logic here, and as a consequence too many fields on AsyncMapOperator. I think it'd be better to generate more of the code in arroyo-sql.